### PR TITLE
2865 update node packages

### LINKED ISSUE #2865: Update npm packages
# Refactor

## Component(s) to be refactored

Some of our npm dependencies are outdated, which means that we are carrying legacy bugs, like [this](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/476)

## Tasks

- [ ] Upgrade major versions of packages, test if everything still works. Packages we should upgrade: (@cakekoa)
  - [x] update all packages (0.25d)

### PR DESCRIPTION
**Pull request by cakekoa**
# What does this PR do?

Fixes #2865.

Updates node packages to newer versions. The biggest change addressed here is updating `react-router-dom` 5.X -> 6.X. Other things to note:
- `babel-eslint` changed to `@babel/eslint-parser`
- `@emotion/core` changed to `@emotion/react`
- `react-json-tree` no longer exports a `JSONTree` by default

A few packages can still be upgraded, but would take more effort:
- `bootstrap` 4.X -> 5.X
- `mui-datatables` 3.X -> 4.X
- `react-table` -> @tanstack/react-table
- `react-tsparticles` 1.X -> 2.X

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/ui/.eslintrc [hunk 1]
```diff
@@ -8,15 +8,14 @@
...
     "ecmaVersion": 6,
-    "sourceType": "module",
-    "allowImportExportEverywhere": false,
     "ecmaFeatures": {
       "jsx": true,
       "globalReturn": false
     },
     "babelOptions": {
-      "configFile": ".babelrc"
+      "configFile": false
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/.eslintrc:18]: so we don't need ".babelrc" anymore? Why is it not deleted then?

File: monkey/monkey_island/cc/ui/.eslintrc [hunk 2]
```diff
@@ -56,10 +65,17 @@
...
+        ]
+      }
+    ],
     "react/no-string-refs": 1,
     "react/display-name": 1,
     "no-useless-escape": 1,
     "no-prototype-builtins": 1
   }
-}
+}
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/.eslintrc:81]: Add newline?

File: monkey/monkey_island/cc/ui/src/components/Main.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ import { doesAnyAgentExist, didAllAgentsShutdown } from './utils/ServerUtils';
 
 let notificationIcon = require('../images/notification-logo-512x512.png');
 
-export const Routes = {
+export const routes = {
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/Main.tsx:34]: Enums are either all capital or CameCase

**Comment by mssalvatore** [comment]
Resolves #3027, #3026, #3025, #3024

**Review by VakarisZ — APPROVED** [review]
Approved, but not tested

**Review by ilija-lazoroski — APPROVED** [review]

**Review by mssalvatore — CHANGES_REQUESTED** [review]
It seems that there's an infinite loop. When the security reports page is opened after and agent runs, the Island's API is bombarded with requests.

**Pull request merged**